qutebrowser/mainwindow/statusbar/percentage.py

```diff
@@ -49,8 +51,8 @@ def set_perc(self, x, y):  # pylint: disable=unused-argument
         elif y is None:
             self.setText('[???]')
         else:
-            text = '[{:02}]' if self.raw else '[{:02}%]'
-            self.setText(text.format(y))
+            text = '[' + str(y).zfill(2) + (']' if self.raw else '%]')
+            self.setText(text)
```

Comment:
Did you actually measure this? :wink:

```
$ python3 -m timeit 'x = 2; "[{:02}%]".format(x)'
1000000 loops, best of 5: 260 nsec per loop

$ python3 -m timeit 'x = 2; str(x).zfill(2) + "%]"'
1000000 loops, best of 5: 256 nsec per loop
```

Surely 4ns for something called once all 100ms doesn't make any difference at all?

Reply:
Hmm, for me it's larger: 
```
[jay@eve ~]$ python3 -m timeit 'x = 2; "[{:02}%]".format(x)'
1000000 loops, best of 5: 203 nsec per loop
[jay@eve ~]$ python3 -m timeit 'x = 2; str(x).zfill(2) + "%]"'
2000000 loops, best of 5: 182 nsec per loop
```

I did work on this before adding the throttling, so it was called a lot more back then (and it was a measurable difference for me at least). I guess it dosen't make a difference now though, so I'll revert it (even though I find this version more readable :P)

Reply:
my main machine:

```sh
python3 -m timeit 'x = 2; "[{:02}%]".format(x)'
```

```
1000000 loops, best of 3: 0.343 usec per loop
```

```sh
python3 -m timeit 'x = 2; str(x).zfill(2) + "%]"'
```

```
1000000 loops, best of 3: 0.459 usec per loop
```

---
another machine:

```sh
python3 -m timeit 'x = 2; "[{:02}%]".format(x)'
```

```
1000000 loops, best of 3: 0.611 usec per loop
```

```sh
python3 -m timeit 'x = 2; str(x).zfill(2) + "%]"'
```

```
1000000 loops, best of 3: 1.23 usec per loop
```

Reply:
Just for reference, I can get it another ~20% faster for me by avoiding the zfill function:
```
[jay@eve qutebrowser]$ python3 -m timeit -s 'x = 2' '"[{:02}%]".format(x)'
1000000 loops, best of 5: 200 nsec per loop
[jay@eve qutebrowser]$ python3 -m timeit -s 'x = 2' 'str(x).zfill(2) + "%]"'
2000000 loops, best of 5: 180 nsec per loop
[jay@eve qutebrowser]$ python3 -m timeit -s 'x = 2' '("%02d" % x) + "%]"'
2000000 loops, best of 5: 143 nsec per loop
[jay@eve qutebrowser]$ 
```

Since we can do it all in one line, I'm just going to take the ~40% performance increase over the original (I think it performs better on older python versions too, which I think the zfill version regressed on). I'm still seeing jitters whenever the statusbar is updated, so I'd like to get the latency we're responsible for down as much as possible.

```
[jay@eve qutebrowser]$ python3 -m timeit -s 'x = 2' '"[%02d%%]" % x'
2000000 loops, best of 5: 136 nsec per loop
[jay@eve qutebrowser]$ python3 -m timeit -s 'x = 2' '"[{:02}%]".format(x)'
1000000 loops, best of 5: 202 nsec per loop
[jay@eve qutebrowser]$ python3.6 -m timeit -s 'x = 2' '"[{:02}%]".format(x)'
1000000 loops, best of 3: 0.495 usec per loop
[jay@eve qutebrowser]$ python3.6 -m timeit -s 'x = 2' '"[%02d%%]" % x'
1000000 loops, best of 3: 0.273 usec per loop
[jay@eve qutebrowser]$
```